### PR TITLE
Added Delete function in Edit Form.

### PR DESCRIPTION
**Pull request by amichard**
#258 
#259 

We already had a delete function before. So it's a matter of putting the confirmation and having it show up in the edit page

Changelog:
- Delete button in Edit Draft
- Minor refactoring
- Changed the modal message and the buttons
- Credit transfers page should no longer show "Cancelled" credit transfers

### CODE REVIEW

File: backend/api/test_credit_trades.py [hunk 1]
```diff
@@ -439,3 +439,46 @@ def test_batch_process(self):
...
+            type=credit_trade_type,
+            number_of_credits=1000,
+            fair_market_value_per_credit=0,
+            zero_reason=credit_trade_zero_reason,
+            trade_effective_date=datetime.datetime.
+            today().strftime('%Y-%m-%d'))
+
+        response = self.gov_client.put('/api/credit_trades/{}/delete'.format(
+            credit_trade.id))
+        assert response.status_code == status.HTTP_200_OK
```
Comment (Sybrand) [inline backend/api/test_credit_trades.py]: Please use self.assertEqual , not assert for unit tests. self.assertEqual will give better error message. Also, more important, when not running in debug mode, these asserts won't work correctly, as relies on __debug__ being True. (See https://docs.python.org/3/reference/simple_stmts.html#assert )

File: backend/api/test_credit_trades.py [hunk 2]
```diff
@@ -439,3 +439,46 @@ def test_batch_process(self):
...
+
+        credit_trade = CreditTrade.objects.get(id=credit_trade.id)
+
+        assert credit_trade.status_id == cancelled_status.id
```
Comment (Sybrand) [inline backend/api/test_credit_trades.py]: Have a look at the "arrange, act, assert" design pattern.

**Comment by amichard** [comment]
[Confirmation Alerts | Delete a Saved Draft of a Credit Transfer |1|](https://trello.com/c/LP1JDWhu/500-confirmation-alerts-delete-a-saved-draft-of-a-credit-transfer-1)

**Comment by amichard** [comment]
[Credit Transfer | Initiator can Delete a credit transfer |3|](https://trello.com/c/PZKSM4Rd/324-credit-transfer-initiator-can-delete-a-credit-transfer-3)

File: backend/api/test_credit_trades.py (hunk 1 above)
Comment (amichard) [reply backend/api/test_credit_trades.py]: Sure. I can change asserts into self.assertEqual instead

**Pull request merged**